src/upgrade/client_request.rs

```diff
@@ -110,11 +110,36 @@ impl Decoder for Codec {
 
         let request_len = status.unwrap();
 
-        let ws_accept = ClientRequest::parse(|name| {
-            let h = headers.iter().find(|h| h.name.eq_ignore_ascii_case(name))?;
-            parse_str(h.value).ok()
-        })?
-        .ws_accept();
+        let mut builder = http::request::Builder::new();
+        if let Some(m) = request.method {
+            let method = http::method::Method::from_bytes(m.as_bytes())
+                .map_err(|e| Error::InvalidRequest(e.into()))?;
+            builder = builder.method(method);
+        }
+        if let Some(uri) = request.path {
+            builder = builder.uri(uri);
+        }
+        match request.version {
+            Some(0) => builder = builder.version(http::Version::HTTP_10),
+            Some(1) => builder = builder.version(http::Version::HTTP_11),
```

Comment:
Hm, interesting. Could you please add this to the server_response part as well if this is being parsed on the other side? We currently hardcode HTTP 1.1 there